aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -71,9 +71,10 @@ void replication_pad1d_out_cpu_template(
   int pad_l = paddingSize[0];
   int pad_r = paddingSize[1];
 
-  TORCH_CHECK(input_.numel() > 0
-      && (input_.ndimension() == 2 || input_.ndimension() == 3),
-      "non-empty 2D or 3D (batch mode) tensor expected for input");
+  // allow empty batch size but not other dimensions.
+  TORCH_CHECK((input_.dim() == 2 && input_.size(0) != 0 && input_.size(1) != 0) ||
+              (input_.dim() == 3 && input_.size(1) != 0 && input_.size(2) != 0),
+              "2D or 3D (batch mode) tensor for input, but got: ", input_.sizes());
```

Comment:
Can you please make error message more descriptive? "Expected 2D or 3D (batch mode) tensor for input with possibly 0 batch dimension and other dimensions non-zero, but got" 
Same for other error messages. Also, you seem to test 2D tensor and expect NotImplementedError there, so is this error message even true?

Reply:
The replication pad 1D function implementation in C++ and CUDA has functionality for 2D tensors but seems to be disabled from the `torch.nn` Python interface.

Reply:
OK I'll change the error messages.